app/models/split.rb

```diff
@@ -82,8 +119,34 @@ def create_decision!(params = {})
     build_decision(params).tap(&:save!)
   end
 
+  def registry
+    if try(:feature_incomplete?)
+      knock_out_weightings(super)
+    else
+      super
+    end
+  end
+
   private
 
+  def knock_out_weightings(registry_hash, to: "false")
+    target_variant = to.to_s
+    knock_out_weightings_if_possible(registry_hash, to: target_variant) || log_knockout_error(target_variant) && registry_hash
+  end
+
+  def knock_out_weightings_if_possible(registry_hash, to:)
+    found_key = false
+    knocked_out_registry = registry_hash.each_with_object({}) do |(k, _), h|
+      h[k] = (k.to_s == to ? (found_key = true && 100) : 0)
+    end
+    found_key ? knocked_out_registry : nil
+  end
+
+  def log_knockout_error(to)
+    logger.error "Failed to knock out weightings of split #{name.inspect} because variant #{to.inspect} not found."
+    true
```

Comment:
Anything that inherits from the stdlib `Logger` should already return `true` here, but I suppose there's always the odd chance that someone's custom logger causes bizarre behavior in the spot where we chain it onto an `&&`.

Reply:
yup, built for safety.